lightning_client.go

```diff
@@ -837,6 +837,44 @@ func newLightningClient(conn *grpc.ClientConn,
 	}
 }
 
+type unlockerClient struct {
+	client lnrpc.WalletUnlockerClient
+}
+
+func newUnlockerClient(conn *grpc.ClientConn) *unlockerClient {
+	return &unlockerClient{
+		client: lnrpc.NewWalletUnlockerClient(conn),
+	}
+}
+
+// unlocked returns a boolean that indicates whether the wallet is currently
+// unlocked.
+func (u *unlockerClient) unlocked(ctx context.Context) bool {
+	// Try to unlock lnd with no password. We expect to receive an error
+	// in all cases:
+	// wallet locked: password is incorrect
+	// wallet unlocked: unlocker not active
+	// no password: unlocker not active
+	// However, we check for a nil error to prevent a panic, we check for
+	// the nil case anyway, and log if we do not fail as expected.
+	_, err := u.client.UnlockWallet(ctx, &lnrpc.UnlockWalletRequest{})
```

Comment:
This is a great idea!
Unfortunately this seems to cause some problems:
 - This will try to unlock the wallet once per second. Because of the scrypt operation, this itself takes multiple seconds and a lot of RAM.
 - There seems to be a deadlock/race when doing the actual unlock. The user call tries to open the wallet DB at the same time this method does. Eventually `lnd` gives up and logs this error: `[ERR] LNWL: Failed to open database: timeout`

I think this can quite easily be fixed by reversing the call: Try to call `GetInfo` and return `true` if `codes.Unimplemented` is no longer returned.